js/build.js

```diff
@@ -8,10 +8,9 @@ Fliplet.Widget.instance('notification-inbox-1-0-0', function (data) {
 
   // Sample implementation to initialize the widget
   var inbox = new NotificationInbox(element, data);
-  Fliplet.Hooks.on('beforeNotificationsInit', function () {
-    // Initialize Notification Inbox component and stops Notifications app component initialization
-    inbox.init();
-    return Promise.reject();
+  Fliplet.Hooks.on('beforeNotificationsInit', function (appComponentData, options) {
```

Comment:
How are we making sure this hook is registered with `.on()` before it is `.run()`?

Reply:
(I feel that I might be asking a stupid question about the fundamentals of promises and hooks here. It's nearly 9pm...)

Reply:
@tonytlwu I haven't changed the order on the `on/run` registration between the two components so it should just keep working as previously

Reply:
If anything, `run` now runs a little later because of these changes, which is good https://github.com/Fliplet/fliplet-widget-notifications/pull/6/files

Reply:
That's good, because I **have** encountered errors in the past from the notification inbox component that says `appNotifications is undefined`, so we do currently have a race condition problem.